lib/repl.js

```diff
@@ -851,14 +851,15 @@ function complete(line, callback) {
           // Exclude versioned names that 'npm' installs.
           continue;
         }
-        if (exts.indexOf(ext) !== -1) {
+        abs = path.resolve(dir, name);
+        isDirectory = fs.statSync(abs).isDirectory();
+        if (!isDirectory && exts.indexOf(ext) !== -1) {
           if (!subdir || base !== 'index') {
             group.push(subdir + base);
           }
         } else {
-          abs = path.resolve(dir, name);
           try {
```

Comment:
The `try` should be moved into the if statement as it is not necessary for non directories.